usr/src/uts/common/fs/zfs/dmu.c

```diff
@@ -1476,6 +1476,44 @@ dmu_return_arcbuf(arc_buf_t *buf)
  * dmu_write().
  */
 void
+dmu_assign_arcbuf_dnode(dnode_t *dn, uint64_t offset, arc_buf_t *buf,
+    dmu_tx_t *tx)
```

Comment:
This function looks like mostly copy/paste of dmu_assign_arcbuf().  Could you change dmu_assign_arcbuf() to call dmu_assign_arcbuf_dnode(), so that we aren't adding duplicated code?